src/BizHawk.Emulation.Cores/Waterbox/NymaCore.cs

```diff
@@ -185,6 +185,31 @@ public abstract unsafe partial class NymaCore : WaterboxCore
 			return t;
 		}
 
+		// inits only to get settings info
+		// should only ever be called if no SettingsInfo cache exists statically within the core
+		protected void InitForSettingsInfo(string wbxFilename)
+		{
+			_nyma = PreInit<LibNymaCore>(new WaterboxOptions
+			{
+				Filename = wbxFilename,
+				// todo: maybe make these smaller? not much is going to occur within this core instance...

```

Comment:
I like this the way it is now:  The exact same memory settings should be used to guarantee a reliable core bootstrap.  All these megabytes don't cost much since they're eliminated soon after, and you'd need that much ram to load the core anyway.

To make this more explicit, could we abstract away this `new WaterboxOptions` call and make it exactly the same one that is used in `DoInit`?

Reply:
Do you mean making a separate function that returns a new WaterboxOptions returning the same options for each? Not sure if that's exactly a good idea given a dummy firmware delegate would probably need to be made. Perhaps best to keep it as is (although todo comment removed).

Reply:
The `WaterboxOptions` class doesn't contain any of the delegate info.

Reply:
Ah, misinterpreted the delegate array as part of WaterboxOptions when that's just the next arg in PreInit, I see.